crates/shared/src/price_estimation/native/oneinch.rs

```diff
@@ -105,16 +125,36 @@ async fn update_prices(
         .text()
         .await
         .context("Failed to fetch Native 1Inch prices")?;
-    let result = serde_json::from_str::<Response>(&response)
-        .with_context(|| format!("Failed to parse Native 1inch prices from {response:?}"))?;
-    Ok(result.0)
+    let prices = serde_json::from_str::<Response>(&response)
+        .with_context(|| format!("Failed to parse Native 1inch prices from {response:?}"))?
+        .0;
+
+    let token_infos = token_info
+        .get_token_infos(&prices.keys().copied().collect::<Vec<_>>())
+        .await;
+
+    let normalized_prices = prices
+        .into_iter()
+        .filter_map(|(token, price)| {
+            let Some(decimals) = token_infos.get(&token).and_then(|info| info.decimals) else {
+                tracing::debug!(?token, "could not fetch decimals; discarding spot price");
+                return None;
+            };
+            let normalized_price = price.to_f64_lossy() / 10f64.powf(decimals.into());
```

Comment:
Delaying the rounding error makes sense. But we have to do the computation with `BigRationals` instead of `U256`. Otherwise we'll lose even more precision given that `U256` can't encode fractional numbers.